File: Programmers/LEVEL_1/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";

    for (int i = 0; i < s.size(); i++)
    {
        if (s[i] == ' ')
            answer += s[i];
        else
        {
            int start = (s[i] >= 'a') ? 'a' : 'A';
            answer += start + (s[i] - start + n) % 26;
        }
    }

    // for(int i=0; i<s.size(); i++)
    // {
    //     char ch=s[i];
    //     if('A'<=ch&&ch<='Z')
    //     {
    //         if(ch+n>'Z')
    //         {
    //             char temp = ch + n - 'Z';
    //             temp += 64;
    //             answer+=temp;
    //         }
    //         else
    //         {
    //             answer += (ch+n);
    //         }
    //     }
    //     else if('a'<=ch&&ch<='z')
    //     {
    //         if(ch+n>'z')
    //         {
    //             char temp = ch + n - 'z';
    //             temp +=96;
    //             answer+=temp;
    //         }
    //         else
    //         {
    //             answer+= (ch+n);
    //         }
    //     }
    //     else if(ch==' ')
    //     {
    //         answer+=ch;
    //     }
    // }
    return answer;
}

File: Programmers/LEVEL_1/시저 암호.js
function solution(s, n) {
  var answer = "";

  let arr = [];
  for (let i in s) {
    let temp = s[i].charCodeAt();
    if (temp == " ") {
      arr.push(32);
      continue;
    }
    // let temp = s[i].charCodeAt()+n;
    if (temp >= 97) {
      if (temp + n > 122) {
        temp = temp + n - 26;
      } else {
        temp += n;
      }
    } else if (temp >= 65) {
      if (temp + n > 90) {
        temp = temp + n - 26;
      } else {
        temp += n;
      }
    }
    arr.push(temp);
  }

  for (let i in arr) {
    answer += String.fromCharCode(arr[i]);
  }
  return answer;
}
